engine/src/main/java/org/entando/entando/aps/system/services/entity/AbstractEntityTypeService.java

```diff
@@ -100,15 +75,25 @@ protected EntityTypeShortDto toDto(IApsEntity src) {
         };
     }
 
-    protected PagedMetadata<EntityTypeShortDto> getShortEntityTypes(String entityManagerCode, RestListRequest requestList) {
+    protected PagedMetadata<EntityTypeShortDto> getShortEntityTypes(String entityManagerCode,
+                                                                    RestListRequest requestList) {
+
         IEntityManager entityManager = this.extractEntityManager(entityManagerCode);
-        List<IApsEntity> entityTypes = new ArrayList<>(entityManager.getEntityPrototypes().values());
+
         Map<String, String> fieldMapping = this.getEntityTypeFieldNameMapping();
-        entityTypes.stream().filter(i -> this.filterObjects(i, requestList.getFilters(), fieldMapping));
-        Collections.sort(entityTypes, new CaseInsensitiveBeanComparator(this.getFieldName(requestList.getSort(), fieldMapping)));
+
+        TransformingComparator caseInsensitiveComparator = createCaseInsensitiveComparator();
         if (!RestListRequest.DIRECTION_VALUE_DEFAULT.equals(requestList.getDirection())) {
-            Collections.reverse(entityTypes);
+            caseInsensitiveComparator.reversed();
         }
+
+        List<IApsEntity> entityTypes = entityManager.getEntityPrototypes()
+                .values()
+                .stream()
+                .filter(i -> this.filterObjects(i, requestList.getFilters(), fieldMapping))
+                .sorted(new BeanComparator<>(getFieldName(requestList.getSort(), fieldMapping), caseInsensitiveComparator))
```

Comment:
@zonia3000 Here I'd rather have used your CaseInsensitiveComparator, but Java was giving me some grief because of the types of your comparator, I replaced with a TransformerComparator in place. Maybe you have a better idea about how to use your comparator in the stream here

Reply:
Hi Rodrigo, `CaseInsensitiveBeanComparator` can't be very typed, so I think that your solution is better. We should remove that class and its test if it is not used anymore.

Reply:
In this case, I'll remove that class and move the new comparator code into a separate class to make it easier to be reused.